modules/ecs-service/main.tf

```diff
@@ -0,0 +1,696 @@
+  role       = aws_iam_role.task_execution.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+
+# ------------------------------------------------------------
+
+# THE FOLLOWING SECTION IS USED TO CONFIGURE THE ECS SERVICE,
+
+# AND ASSOCIATED RESOURCES INCLUDING THE ALB, AUTOSCALING,
+
+# SERVICE CONNECT.
+
+# ------------------------------------------------------------
+
+
+# -------------------------------------------
+# CREATE THE SERVICE CONNECT RANDOM ID
+# -------------------------------------------
+
+# Namespaces the CloudMap services to avoid name conflicts
+# with other services sharing the same Service Connect namespace.
+resource "random_id" "service_connect" {
+  count = !var.create_scheduled_task && var.enable_service_connect ? 1 : 0
+
+  byte_length = 8
+}
+
+
+# -------------------------------------------
+# CREATE THE ECS SERVICE
+# -------------------------------------------
+
+locals {
+  # The ECS service role is required when using an
+  # Application Load Balancer with the ECS service.
+  aws_ecs_service_role = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/aws-service-role/ecs.amazonaws.com/AWSServiceRoleForECS"
+}
+
+resource "aws_ecs_service" "service" {
+  count = !var.create_scheduled_task ? 1 : 0
+
+  cluster = data.aws_ecs_cluster.cluster.arn
+
+  name            = var.ecs_service_name
+  task_definition = local.task_definition
+
+  iam_role = var.enable_load_balancer ? local.aws_ecs_service_role : null
+
+  deployment_circuit_breaker {
+    enable   = var.enable_deployment_rollback
+    rollback = true
+  }
+
+  # The upper and lower limits (as a percentage of the service's desiredCount)
+  # of the unmber of running tasks that should be maintained in the service
+  # during a deployment.
+  deployment_maximum_percent         = 200
+  deployment_minimum_healthy_percent = 50
+  desired_count                      = var.desired_number_of_tasks
+
+  enable_ecs_managed_tags = true
+  enable_execute_command  = false
+  force_new_deployment    = true
+
+  health_check_grace_period_seconds = var.enable_load_balancer ? 0 : null
+
+  dynamic "load_balancer" {
+    for_each = var.enable_load_balancer ? [1] : []
+
+    content {
+      container_name   = var.ecs_service_name
+      container_port   = var.ecs_container_port
+      target_group_arn = aws_lb_target_group.alb[0].arn
+    }
+  }
+
+  # Tasks are placed on container instances so as to leave the
+  # least amount of unused CPU or memory. This strategy minimizes
+  # the number of container instances in use.
+  ordered_placement_strategy {
+    type  = "binpack"
+    field = "memory"
+  }
+
+  service_connect_configuration {
+    enabled = var.enable_service_connect
+    service {
+      port_name = random_id.service_connect[0].hex
+      client_alias {
+        port     = var.ecs_container_port
+        dns_name = var.ecs_service_name
+      }
+    }
+  }
+
+  timeouts {
+    delete = "30m"
```

Comment:
30 minute timeout?

Reply:
Yes, just going beyond the time it should actually take to give a buffer. Deregistration can take some time if its connected to a load balancer and times are changed. Especially if there are a lot of tasks.